src/NationalInstruments.Analyzers/Correctness/ApprovedNamespaceCodeFixProvider.cs

```diff
@@ -33,18 +37,55 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 var namespaceName = (await location.SourceTree.GetRootAsync(context.CancellationToken).ConfigureAwait(false))
                     .FindNode(location.SourceSpan)
                     .ToString();
-                var codeAction = CodeAction.Create(
-                    Resources.NI1800_CodeFixTitle,
-                    cancellationToken => AddNamespaceAsync(context, namespaceName),
-                    nameof(ApprovedNamespaceCodeFixProvider));
-                context.RegisterCodeFix(codeAction, context.Diagnostics);
+                var approvedNamespacesFilePaths = diagnostic.Properties.Values;
+                foreach (var path in approvedNamespacesFilePaths)
```

Comment:
This is just future proofing. We plan to have multiple scoped files going forward.